test/KdTreeTest.hs

```diff
@@ -59,6 +59,14 @@ prop_removePreservesInvariant points pKill =
     Kd.allSubtreesAreValid $ tree `Kd.remove` pKill
     where tree = Kd.fromList points
 
+prop_removeEdgeTest::Int->Bool
+prop_removeEdgeTest dummy = ret
+  where
+    a = Kd.Point3d 2 2 2
+    b = Kd.Point3d 2 3 3 --Change this line to Kd.Point3d 3 3 3,Kd.remove will work.
+    kdTree = Kd.fromList [a,b]
+    ret = ((Kd.remove kdTree b) /= kdTree)--return if Kd.remove works.
```

Comment:
You might want to express this as `Kd.remove kdTree b == Kd.fromList [a]`. Otherwise `remove` could return an empty tree and this test would still pass.

Reply:
Yes. I wanted to express Kd.remove doesn't remove b in this case.